mpsse_darwin.go

```diff
@@ -339,18 +321,20 @@ func (m *Mpsse) Start() error {
 	return nil
 }
 
-
 // Write sends data out via the selected serial protocol.
 //
 // It is a wrapper for the mpsse C function:
 //     int Write(struct mpsse_context *mpsse, char *data, int size);
 func (m *Mpsse) Write(data string) error {
 	dataP := C.CString(data)
-	defer C.free(unsafe.Pointer(dataP))
+	status := int(C.Write(m.ctx, dataP, C.int(len(data))))
+
+	// free whether we succeed or fail.
+	// DANGER: This free used to be called in a defer statement.
+	// The code was called because a trace in the code was emitted,
+	// but the memory was _not_ freed. Suspect a compiler bug.
+	C.free(unsafe.Pointer(dataP))
 
-	// FIXME -- need to check that this works. not clear that len(data) gives
-	// us the size that we want. maybe unsafe.Sizeof will give the int
-	// size we want? but I'm also unsure about that. will need to test.
 	status := int(C.Write(m.ctx, dataP, C.int(len(data))))
```

Comment:
should this be removed? It looks like this line has moved up to line 330

Reply:
Ugh - yes - thanks!